### PR TITLE
DOC: update the pandas.DataFrame.plot.area docstring

### PR DESCRIPTION
**Pull request by DZPM**
- [x] PR title is "DOC: update the pandas.DataFrame.plot.area docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py pandas.DataFrame.plot.area`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single pandas.DataFrame.plot.area`
- [x] It has been proofread on language by another sprint participant

```
################################################################################
#################### Docstring (pandas.DataFrame.plot.area) ####################
################################################################################

Draw an area plot of the input series using matplotlib.

An area plot displays graphically quantitative data.
This function wraps the matplotlib area function.

Parameters
----------
x : label or position, optional
    Coordinates for X point.
y : label or position, optional
    Coordinates for Y point.
kwds : optional
    Keyword arguments to pass on to :py:meth:`pandas.DataFrame.plot`.

Returns
-------
matplotlib.AxesSubplot or numpy.ndarray
    Area plot generated.

See Also
--------
pandas.DataFrame.plot : Draw a plot using matplotlib.

Examples
--------
.. plot::
    :context: close-figs

    This example draws an area plot based on the length and width of
    some animals, displayed in three bins

    >>> df = pd.DataFrame({
    ...     'sales': [3, 2, 3, 9, 10, 6],
    ...     'visits': [20, 42, 28, 62, 81, 50],
    ... })
    >>> area = df.plot.area()

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DataFrame.plot.area" correct. :)
```

### CODE REVIEW

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -2818,18 +2818,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
 
     def area(self, x=None, y=None, **kwds):
         """
-        Area plot
+        Draw an area plot of the input series using matplotlib.
+
+        An area plot displays graphically quantitative data.
```
Comment (kynan) [inline pandas/plotting/_core.py]: displays quantitative data visually

File: pandas/plotting/_core.py [hunk 2]
```diff
@@ -2818,18 +2818,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+        This function wraps the matplotlib area function.
 
         Parameters
         ----------
-        x, y : label or position, optional
-            Coordinates for each point.
-        `**kwds` : optional
+        x : label or position, optional
+            Coordinates for X point.
```
Comment (kynan) [inline pandas/plotting/_core.py]: X axis

File: pandas/plotting/_core.py [hunk 3]
```diff
@@ -2818,18 +2818,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+        y : label or position, optional
+            Coordinates for Y point.
```
Comment (kynan) [inline pandas/plotting/_core.py]: Y axis

File: pandas/plotting/_core.py [hunk 4]
```diff
@@ -2818,18 +2818,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
 
         Returns
         -------
-        axes : matplotlib.AxesSubplot or np.array of them
+        matplotlib.AxesSubplot or numpy.ndarray
+            Area plot generated.
+
+        See Also
+        --------
+        pandas.DataFrame.plot : Draw a plot using matplotlib.
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: The explanation here is a bit strange, as it would seem to say that this one is not plotting with matplotlib

File: pandas/plotting/_core.py [hunk 5]
```diff
@@ -2818,18 +2818,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+        y : label or position, optional
+            Coordinates for Y point.
+        kwds : optional
             Keyword arguments to pass on to :py:meth:`pandas.DataFrame.plot`.
 
         Returns
         -------
-        axes : matplotlib.AxesSubplot or np.array of them
+        matplotlib.AxesSubplot or numpy.ndarray
+            Area plot generated.
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: Can you explain something about when ndarray is returned?

File: pandas/plotting/_core.py [hunk 6]
```diff
@@ -2818,18 +2818,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
 
     def area(self, x=None, y=None, **kwds):
         """
-        Area plot
+        Draw an area plot of the input series using matplotlib.
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: "input series" -> input is a bit strange wording as it is the calling object (not passed as input) + it should be DataFrame instead of Series ?

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks! added some comments

**Comment by dukebody** [comment]
Can you post a screenshot of the rendered HTML? It will help reviewing without having to check out your branch and compiling it.

File: pandas/plotting/_core.py (hunk 6 above)
Comment (dukebody) [reply pandas/plotting/_core.py]: I don't know if we want to mention the underlying library this is using at this level. Does the user really care so much? In the extended description is more OK IMO.

Comment (dukebody) [reply pandas/plotting/_core.py]: Also, as @jorisvandenbossche says, the description doesn't make sense for dataframes. Probably what the function does is to display a stacked area plot?

File: pandas/plotting/_core.py (hunk 1 above)
Comment (dukebody) [inline pandas/plotting/_core.py]: It would be great to add some info about why this function is useful or some basic use cases. `Displays quantitative data visually` doesn't really give me very interesting information about what is the shape or usage of the plot.

File: pandas/plotting/_core.py [hunk 7]
```diff
@@ -2818,18 +2818,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+
+        An area plot displays graphically quantitative data.
+        This function wraps the matplotlib area function.
 
         Parameters
         ----------
-        x, y : label or position, optional
-            Coordinates for each point.
-        `**kwds` : optional
+        x : label or position, optional
```
Comment (dukebody) [inline pandas/plotting/_core.py]: This should follow the guide: https://python-sprints.github.io/pandas/guide/pandas_docstring.html#section-3-parameters

So it should indicate the type of the parameter in the first line (plus optional), and later the meaning of the parameter.

File: pandas/plotting/_core.py (hunk 2 above)
Comment (dukebody) [inline pandas/plotting/_core.py]: From this description of the arguments `x` and `y` I really don't understand what they change in the plot. Can we make the parameter description a bit longer explaining what do they mean?

File: pandas/plotting/_core.py [hunk 8]
```diff
@@ -2818,18 +2818,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+            Coordinates for X point.
+        y : label or position, optional
+            Coordinates for Y point.
+        kwds : optional
```
Comment (dukebody) [inline pandas/plotting/_core.py]: As mentioned in other PRs, you should use `**kwds` instead. Sorry for the great big confusion here.

File: pandas/plotting/_core.py (hunk 5 above)
Comment (dukebody) [reply pandas/plotting/_core.py]: Yep. In this case the existing explanation `Area plot generated` doesn't tell a lot. How is a `numpy.ndarray` the area plot generated?

File: pandas/plotting/_core.py [hunk 9]
```diff
@@ -2818,18 +2818,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+        See Also
+        --------
+        pandas.DataFrame.plot : Draw a plot using matplotlib.
+
+        Examples
+        --------
+        .. plot::
+            :context: close-figs
+
+            This example draws an area plot based on the length and width of
```
Comment (dukebody) [inline pandas/plotting/_core.py]: 1) I think is better to put the text before the start of the `plot` block, and leave the block only for the code that constructs the plot.
2) The example this sentence talks about doesn't correspond to the dataframe below.
3) Don't write in 3rd person `This example does this`. Better write directly in imperative or present form, like `Draw an area plot based on` or `We can draw an area plot based on...`, but IMO mentioning `this example` is somewhat redundant when we are already in the examples section. :)

File: pandas/plotting/_core.py [hunk 10]
```diff
@@ -2818,18 +2818,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+            some animals, displayed in three bins
```
Comment (dukebody) [inline pandas/plotting/_core.py]: In three bins? Where does the three come from?

File: pandas/plotting/_core.py [hunk 11]
```diff
@@ -2818,18 +2818,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+
+            >>> df = pd.DataFrame({
+            ...     'sales': [3, 2, 3, 9, 10, 6],
+            ...     'visits': [20, 42, 28, 62, 81, 50],
+            ... })
+            >>> area = df.plot.area()
```
Comment (dukebody) [inline pandas/plotting/_core.py]: I'm not sure about this, but if the result of this method is an axis object, perhaps it's better to assign it to a variable named `ax` like in other examples.

Can you add other examples where the usage of the `x` and `y` parameters is shown? Also an example where a `numpy.ndarray` is returned.

**Review by dukebody — CHANGES_REQUESTED** [review]
I believe this needs some better explanations of the extended description, arguments and more examples that help to understand how this graph works.

**Comment by dukebody** [comment]
Is there anything we can do to help finishing the work on this PR?

**Comment by DZPM** [comment]
I'll update the PR later this week, just need some time :+1:

File: pandas/plotting/_core.py (hunk 8 above)
Comment (DZPM) [reply pandas/plotting/_core.py]: Using `**kwds` raises errors in the current version of the validator:

	Errors in parameters section
		Parameters {'kwds'} not documented
		Unknown parameters {'**kwds'}

I think it's important to be consistent between the validator and the code.

Of course, once the validator is updated, I agree this must be changed.

File: pandas/plotting/_core.py (hunk 10 above)
Comment (DZPM) [reply pandas/plotting/_core.py]: Yes, it's wrong, copied it from another example. Removed it.

File: pandas/plotting/_core.py (hunk 11 above)
Comment (DZPM) [reply pandas/plotting/_core.py]: Not sure about how to define a "usable" example, a little help would be appreciated!

**Comment by DZPM** [comment]
Fixed most of the suggestions, except for:
- writing `**kwds` instead of `kwds`. That would break the validation.
- more/better examples: I ask for inspiration :)

Also, a screenshot of the example:
image

File: pandas/plotting/_core.py (hunk 8 above)
Comment (jorisvandenbossche) [reply pandas/plotting/_core.py]: You can ignore that error in the validation script for now.

File: pandas/plotting/_core.py [hunk 12]
```diff
@@ -3115,19 +3115,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
 
     def area(self, x=None, y=None, **kwds):
         """
-        Area plot
+        Draw an stacked area plot.
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: an -> a

File: pandas/plotting/_core.py (hunk 8 above)
Comment (DZPM) [reply pandas/plotting/_core.py]: Ok, updated!

File: pandas/plotting/_core.py [hunk 13]
```diff
@@ -3115,19 +3115,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+
+        An area plot displays quantitative data visually.
+        This function wraps the matplotlib area function.
 
         Parameters
         ----------
-        x, y : label or position, optional
-            Coordinates for each point.
-        `**kwds` : optional
+        x : label or position, optional, default None
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: you can leave out the ", default None" part (saying it is "optional" should be informative enough"). 

But we should specify here what it used for `x` by default. I suppose this is the index?

(and the same comment for `y`

File: pandas/plotting/_core.py [hunk 14]
```diff
@@ -3115,19 +3115,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+            Coordinates for X axis.
+        y : label or position, optional, default None
+            Coordinates for Y axis.
+        **kwds : optional
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: Can you also document the `stacked` keyword? I know it is not in the signature, but it is a relevant one for this type of plot, so I would add it. 
See http://pandas.pydata.org/pandas-docs/stable/visualization.html#area-plot

(the validation script will complain about it, but you can ignore that again)

File: pandas/plotting/_core.py [hunk 15]
```diff
@@ -3115,19 +3115,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
             Additional keyword arguments are documented in
             :meth:`pandas.DataFrame.plot`.
 
         Returns
         -------
-        axes : :class:`matplotlib.axes.Axes` or numpy.ndarray of them
+        matplotlib.axes.Axes or numpy.ndarray
+            Area plot, or array of area plots.
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: It would be good to add when an array is returned, which is when `subplots=True` is specified

File: pandas/plotting/_core.py [hunk 16]
```diff
@@ -3115,19 +3115,42 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+        Draw an area plot based on the length and width of some animals:
+
+        .. plot::
+            :context: close-figs
+
+            >>> df = pd.DataFrame({
+            ...     'sales': [3, 2, 3, 9, 10, 6],
+            ...     'visits': [20, 42, 28, 62, 81, 50],
+            ... })
+            >>> ax = df.plot.area()
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: Some ideas for the examples:

- I would also show one where you use the `x` and `y` keywords
- I would show one with `stacked=False`
- I think it is typical to use area plots with time series data. So maybe it would be nice to add a DatetimeIndex to the `df`.

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the updates! Added some more comments

File: pandas/plotting/_core.py (hunk 13 above)
Comment (TomAugspurger) [reply pandas/plotting/_core.py]: Yes I believe it's the index by default.

**Comment by TomAugspurger** [comment]
> writing **kwds instead of kwds. That would break the validation.

You can ignore this, `**kwargs` is correct, the script is incorrect :)

**Comment by TomAugspurger** [comment]
http://pandas.pydata.org/pandas-docs/stable/visualization.html#area-plot has an example. The same thing with a datetimeidnex from `pd.date_range` would be nice (maybe a bit wider of an aspect ratio).

**Comment by DZPM** [comment]
Hi,

I've updated with all the proposals, and rebased.

Screenshots of the rendered example section:
image
image
image
image
image

It looks much better :) Thanks for the feedback!

**Comment by DZPM** [comment]
Ping @kynan @jorisvandenbossche @TomAugspurger @dukebody 

Feel free to keep reviewing, or do the merge :)

**Comment by jorisvandenbossche** [comment]
@DZPM Thanks for the reminder! 
Can you add a datetime index to the example frame? See https://github.com/pandas-dev/pandas/pull/20170#issuecomment-373364136

File: pandas/plotting/_core.py [hunk 17]
```diff
@@ -3188,19 +3188,85 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+        An area plot displays quantitative data visually.
+        This function wraps the matplotlib area function.
 
         Parameters
         ----------
-        x, y : label or position, optional
-            Coordinates for each point.
-        `**kwds` : optional
+        x : label or position, optional
+            Index, coordinates for X axis.
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: The "Index, " part is a bit confusing I think. I would reword this as : `Coordinates for the Y axis. By default uses the index.`

File: pandas/plotting/_core.py [hunk 18]
```diff
@@ -3188,19 +3188,85 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+        y : label or position, optional
+            Index, coordinates for Y axis.
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: Here the default is not the index, but all columns. So maybe something like "Column to plot. By default uses all columns." ?

File: pandas/plotting/_core.py [hunk 19]
```diff
@@ -3188,19 +3188,85 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+        .. plot::
+            :context: close-figs
+
+            >>> df = pd.DataFrame({
+            ...     'sales': [3, 2, 3, 9, 10, 6],
+            ...     'signups': [5, 5, 6, 12, 14, 13],
+            ...     'visits': [20, 42, 28, 62, 81, 50],
+            ... })
+            >>> ax = df.plot.area(y='sales')
+            >>> ax = df.plot.area(y='signups')
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: I think showing only the first is fine to illustrate the `y` keyword (to save some vertical screen estate)

(if you want to have all of them on a separate figure, one could also use `subplots=True`)

**Review by jorisvandenbossche — COMMENTED** [review]
Added a few more comments

BTW, for future reference, if you add new commits to the branch with additional changes instead of squashing everything into a single commit, it is a bit easier to see what has been updated since the last review

**Comment by datapythonista** [comment]
@DZPM can you please update this PR based on the previous feedback

**Comment by DZPM** [comment]
Sure, will do. (Sorry, I've been busy and the PR slipped my mind...)

**Comment by DZPM** [comment]
Done. Added the changes in a new commit, and rebased.

File: pandas/plotting/_core.py [hunk 20]
```diff
@@ -3337,19 +3337,87 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+
+        .. plot::
+            :context: close-figs
+
+            >>> df = pd.DataFrame({
+            ...     'sales': [3, 2, 3, 9, 10, 6],
+            ...     'signups': [5, 5, 6, 12, 14, 13],
+            ...     'visits': [20, 42, 28, 62, 81, 50],
+            ... }, index=pd.date_range(start='2018/01/01', end='2018/07/01',
+            ...                        freq='M'))
```
Comment (datapythonista) [inline pandas/plotting/_core.py]: is the `.. plot::` directive resetting the namespace? Otherwise, It'd be good to reuse the DataFrame instead of repeating it in every example.

File: pandas/plotting/_core.py [hunk 21]
```diff
@@ -3337,19 +3337,87 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
 
         Returns
         -------
-        axes : :class:`matplotlib.axes.Axes` or numpy.ndarray of them
+        matplotlib.axes.Axes or numpy.ndarray
+            Area plot, or array of area plots if subplots is True
+
+        See Also
+        --------
+        pandas.DataFrame.plot : Draw plots.
```
Comment (datapythonista) [inline pandas/plotting/_core.py]: You can get rid of `pandas.` and just leave `DataFrame.plot`. Something a bit more descriptive than "Draw plots." could be useful.

Not sure if in similar docstrings we also added the matplotlib equivalent method. Can you check the docstring for `DataFrame.kde` for example, and be consistent with it?

File: pandas/plotting/_core.py [hunk 22]
```diff
@@ -3337,19 +3337,87 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+        This function wraps the matplotlib area function.
 
         Parameters
         ----------
-        x, y : label or position, optional
-            Coordinates for each point.
-        `**kwds` : optional
+        x : label or position, optional
+            Coordinates for the X axis. By default uses the index.
+        y : label or position, optional
```
Comment (datapythonista) [inline pandas/plotting/_core.py]: As @dukebody says, for both `x` and `y`, in the first line we should have the type, more than a short description. It's a bit tricky, as `label` can be anything, but I think we're using something like `object or int, optional`. Then, we can sure clarify that it's the `label` or the `position` in the description.

File: pandas/plotting/_core.py [hunk 23]
```diff
@@ -3337,19 +3337,87 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+            Column to plot. By default uses all columns.
+        stacked : boolean, default True
```
Comment (datapythonista) [inline pandas/plotting/_core.py]: Happy to leave like this now, but as `stacked` is not in the signature, may be we should add it there, or document it under `kwds`. But if we leave it like this, can you replace `boolean` by the Python type `bool` please?

**Review by datapythonista — COMMENTED** [review]
Looks great, just few minor things. Thanks for the changes!

File: pandas/plotting/_core.py (hunk 20 above)
Comment (DZPM) [reply pandas/plotting/_core.py]: Had some issues with the namespace, so I kept it in different places.

Don't have the full environment right now, can't check if it also works the other way.

File: pandas/plotting/_core.py (hunk 21 above)
Comment (DZPM) [reply pandas/plotting/_core.py]: Added the `DataFrame.plot` description.

File: pandas/plotting/_core.py (hunk 22 above)
Comment (DZPM) [reply pandas/plotting/_core.py]: I kept it consistent with  `DataFrame.plot`, which is the more complete docstring example I could find.

**Comment by DZPM** [comment]
Changes done, and rebased. Hope it's ok!

**Comment by TomAugspurger** [comment]
LGTM, thanks!

If you want, check the doc build when it finishes (probably a few hours).

File: pandas/plotting/_core.py (hunk 20 above)
Comment (jorisvandenbossche) [reply pandas/plotting/_core.py]: Recreating the dataframe each time should not be needed

File: pandas/plotting/_core.py [hunk 24]
```diff
@@ -3337,19 +3337,87 @@ def kde(self, bw_method=None, ind=None, **kwds):
...
+
+        .. plot::
+            :context: close-figs
+
+            >>> df = pd.DataFrame({
+            ...     'sales': [3, 2, 3],
+            ...     'visits': [20, 42, 28],
+            ...     'day': ['Monday', 'Tuesday', 'Wednesday'],
+            ... }, index=pd.date_range(start='2018/01/01', end='2018/07/01',
+            ...                        freq='M'))
```
Comment (jorisvandenbossche) [inline pandas/plotting/_core.py]: This example is not running (the index is not correct). And also the labels to not show on the actual plot once this is fixed.

**Review by jorisvandenbossche — COMMENTED** [review]
Some follow-up in https://github.com/pandas-dev/pandas/pull/22307
